crates/services/p2p/src/p2p_service.rs

```diff
@@ -271,6 +271,15 @@ impl FuelP2PService {
         }
     }
 
+    pub fn log_metrics<T>(&self, cb: T)
```

Comment:
Can we use a more descriptive name for `cb` here? Is it a callback? What do we assume the callback does?

Reply:
Also, I guess we're not really logging anything here. Given how this is used, I'd suggest renaming the function to something like `update_metrics`, where the callback could be named `update_function`. I'm open to other names as well of course.

Reply:
good point about the naming, will address.
the callback should be something that updates metrics

Reply:
addressed in 22f7693a5